src/psyclone/tests/psyir/symbols/symboltable_test.py

```diff
@@ -2064,3 +2062,59 @@ def test_resolve_imports_common_symbol(fortran_reader, tmpdir, monkeypatch,
     for dependency in dependency_order:
         symtab.resolve_imports([symtab.lookup(dependency)])
     assert symtab.lookup("common_import").datatype.intrinsic.name == "INTEGER"
+
+
+def test_scope():
+    '''
+    Test that the scope property returns the SymbolTable associated node.
+    '''
+    symtab = SymbolTable()
+    assert symtab.scope is None
+
+    schedule = Schedule()
+    assert schedule.symbol_table.scope is schedule
+
+
+def test_detach():
+    ''' Test that the detach method of a symbol table detaches itself from its
+    current scope and returns itself. '''
+
+    # Create a symbol_table associated a a scope
+    sym_table = SymbolTable()
+    scope = Schedule(symbol_table=sym_table)
+
+    assert sym_table._node is scope
+
+    # Detach the symbol table
+    assert sym_table.detach() is sym_table
+    assert sym_table._node is not scope
+
+
+def test_attach():
+    ''' Test that the attach method bounds a symboltable and a ScopignNode
```

Comment:
s/bounds/binds/ and s/ScopignNode/ScopingNode/

Reply:
Ups.. fixed